@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -837,6 +845,65 @@ export class VisibleTileSet {
         tile.dispose();
     }
 
+    /**
+     * Disposes of tiles that overlap twice and that come from a [[DataSource]] where the [[Tiles]]s
+     * are fully covering.
+     **/
+    private removeDuplicateFullyCoveringTiles(dataSource: DataSource, tile: Tile) {
+        if (this.options.projection.type === ProjectionType.Spherical) {
+            // HARP-7899, currently the globe has no background planes in the tiles (it relies on
+            // the BackgroundDataSource), because the LOD mismatches, hence disabling for globe.
+            return;
+        }
+        if (dataSource.isFullyCovering()) {
+            const ts = dataSource.getTilingScheme();
+            let map = this.m_coveringMap.get(ts)!;
+            if (map === undefined) {
+                map = new Map<number, Tile>();
+                this.m_coveringMap.set(ts, map);
+            }
+            const key = TileOffsetUtils.getKeyForTileKeyAndOffset(tile.tileKey, tile.offset);
+            const entry = map.get(key);
+            if (entry === undefined) {
+                map.set(key, tile);
+            } else {
+                const disposeTile = (toDispose: Tile) => {
```

Comment:
I don't like the idea that we modify tiles that are in the cache. Especially when thinking about the fallback logic it could happen that we fallback to that tile and suddenly it's empty.
Maybe it's good enough to skip the rendering of the covered tile.

Reply:
I don't see the difference, if you fallback to an empty tile (it is disposed), or you fall back to a tile which isn't rendered, the net (visual) result is the same.